Issue #25 - adding more meaningful ObjectMetadata to MkOcket

prondzyn opened the pull request:
Prepared instead of closed #36 pull request. Travis CI build passed now.

prondzyn commented:
@caarlos0 here you have pull request with fixed Travis build. CC: @dmarkov

dmarkov commented:
@prondzyn Let me find a reviewer for this pull request, thanks for submitting it

dmarkov commented:
@pinaf please review this PR

caarlos0 commented:
@prondzyn why did you close #36 ?

prondzyn commented:
@caarlos0 because #37 contains #36 plus Travis build fix so I thought you will be automatically assigned to #37. I understand I was wrong.

pinaf commented on src/test/java/com/jcabi/s3/mock/MkOcketTest.java:
```diff
@@ -0,0 +1,74 @@
+        final Ocket ocket = bucket.ocket("hello.txt");
+        final String text = "hello, world!";
+        new Ocket.Text(ocket).write(text);
+        final Ocket readOcket = new Ocket.Text(bucket.ocket(ocket.key()));
```
@prondzyn let's rename this ocket to `read` and the other ocket to `write`.

pinaf commented on src/test/java/com/jcabi/s3/mock/MkOcketTest.java:
```diff
@@ -0,0 +1,74 @@
+        final Ocket readOcket = new Ocket.Text(bucket.ocket(ocket.key()));
+        final ObjectMetadata metadata = readOcket.meta();
+        Assert.assertEquals("text/plain", metadata.getContentType());
+        Assert.assertEquals(text.length(), metadata.getContentLength());
```
@prondzyn no test for the new `Headers.DATE` key ?

pinaf commented:
@prondzyn 2 comments above

pinaf commented:
@prondzyn what is the purpose of the MIME file? Is it used by the s3 library?

prondzyn commented:
@pinaf 

I changed the names and added test to get Date from metadata.

The mime.types file is used by MimetypesFileTypeMap in MkOcket when no mime types found in standard system directories. Absence of this file broke the Travis [build 122](https://travis-ci.org/jcabi/jcabi-s3/builds/81857432). I added comment to the mime.types file to give a hint where it is used.

pinaf commented on src/test/java/com/jcabi/s3/mock/MkOcketTest.java:
```diff
@@ -0,0 +1,105 @@
+    @Test
+    public void readDateFromMetadata() throws Exception {
+        final Ocket read = new Ocket.Text(this.bucket.ocket(this.write.key()));
+        final ObjectMetadata metadata = read.meta();
```
@prondzyn let's inline this variable

prondzyn replied:
@pinaf do you think about such solution?
final ObjectMetadata metadata =
            new Ocket.Text(this.bucket.ocket(this.write.key())).meta();

pinaf replied:
@prondzyn The first option works. Another one would be this:
```java
final ObjectMetadata metadata = new Ocket.Text(
   this.bucket.ocket(this.write.key())
).meta();
```

prondzyn replied:
@pinaf your looks better. Thanks!

pinaf replied:
@prondzyn there is a chance qulice complains (not sure if that bug was resolved it) so you might have to do this at the end - yeah, it looks weird :)
```java
final ObjectMetadata metadata = new Ocket.Text(
   this.bucket.ocket(this.write.key())
)
    .meta();
```

pinaf commented on src/test/java/com/jcabi/s3/mock/MkOcketTest.java:
```diff
@@ -0,0 +1,105 @@
+     */
+    @Test
+    public void readDateFromMetadata() throws Exception {
+        final Ocket read = new Ocket.Text(this.bucket.ocket(this.write.key()));
```
@prondzyn let's inline this variable

pinaf commented on src/test/java/com/jcabi/s3/mock/MkOcketTest.java:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    /**
+     * Date can be read from MkOcket metadata.
```
@prondzyn `MkOcket can read date from metadata`

pinaf commented on src/test/java/com/jcabi/s3/mock/MkOcketTest.java:
```diff
@@ -0,0 +1,105 @@
+     * @throws Exception If fails
+     */
+    @Test
+    public void readDateFromMetadata() throws Exception {
```
@prondzyn `readsDateFromMetadata`

pinaf commented on src/test/java/com/jcabi/s3/mock/MkOcketTest.java:
```diff
@@ -0,0 +1,105 @@
+     * @throws Exception If fails
+     */
+    @Test
+    public void readContentTypeAndLengthFromMetadata() throws Exception {
```
@prondzyn `readsContentTypeAndLengthFromMetadata`

pinaf commented on src/test/java/com/jcabi/s3/mock/MkOcketTest.java:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    /**
+     * Content Type and Length can be read from MkOcket metadata.
```
@prondzyn `MkOcket can read Content Type and Length from metadata`

pinaf commented on src/test/java/com/jcabi/s3/mock/MkOcketTest.java:
```diff
@@ -0,0 +1,105 @@
+     */
+    @Before
+    public void setUp() throws Exception {
+        final Region region = new MkRegion(this.temp.newFolder());
```
@prondzyn let's inline this variable

pinaf commented:
@prondzyn thanks

pinaf commented:
@prondzyn a few more minor comments and we are good to merge

prondzyn commented:
@pinaf code cleaned up

pinaf commented:
@prondzyn excellent! thanks!

pinaf commented:
@rultor merge

rultor commented:
> @rultor merge

@pinaf Thanks for your request. @yegor256 Please confirm this.

yegor256 commented:
@rultor try to merge pls

rultor commented:
> @rultor try to merge pls

@yegor256 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/5247-143358616)

rultor commented:
> @rultor try to merge pls

@yegor256 Done! FYI, the full log is [here](http://www.rultor.com/t/5247-143358616) (took me 6min)

dmarkov commented:
@pinaf **21 mins** sent to your balance (ID `AP-1TR56345YP2041344`), many thanks! It took 90 hours and 23 mins.... extra minutes for review comments (c=6)... +21 added to your rating, at the moment it is: [+4217](http://www.netbout.com/b/35128?open=rating)

dmarkov commented:
@rultor deploy now pls

rultor commented:
> @rultor deploy now pls

@dmarkov OK, I'll try to deploy now. You can check the progress [here](http://www.rultor.com/t/5247-143970403)

rultor commented:
> @rultor deploy now pls

@dmarkov Done! FYI, the full log is [here](http://www.rultor.com/t/5247-143970403) (took me 7min)
